### PR TITLE
WF CommitTransaction in a separate WF Feature (Lombiq Technologies: OCORE-47)

### LINKED ISSUE #6413: The Commit Transaction Workflows task can be used to circumvent validation
The Commit Transaction task of Workflows will, well, commit the DB transaction immediately. This can be used to commit the transaction even in a state where normally it wouldn't be committed. Depending on what kind of Workflows events are or will be available this can be used to put the application into a corrupted state. This includes at least the ability to circumvent user input validation for content items but opens up the possibility of any other state corruption with security implications as well.

Simple demonstration:

image

Now when you save a Blog Post with a validation error the changes will still be persisted to the database. Unless those users who can edit workflows can't be completely trusted (both to not be malicious and not to make mistakes) this seems very risky. This can be a problem in any setup but will be one in every SaaS one.

My proposal would be to at least put the Commit Transaction task (and any risky future ones) into its separate feature (or module) so they can be disabled completely.

Otherwise, I'm not sure I see where this task would be useful. Where would you utilize such a low-level operation when creating a workflow? Even as a developer you have to think twice when committing the ambient transaction explicitly so it seems to me that surfacing this on the admin UI is dangerous.

**Verdict:** So as we discussed:

- For now, we can put the Commit Transaction task into its own feature so its usage can be easily prevented if you don't need it.
- https://github.com/OrchardCMS/OrchardCore/issues/6455

### PR DESCRIPTION
**Pull request by jtkech**
Fixes #6413 

See my last comment in the related issue https://github.com/OrchardCMS/OrchardCore/issues/6413#issuecomment-803674683

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Workflows/Manifest.cs [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
 [assembly: Feature(
     Id = "OrchardCore.Workflows.Http",
-    Name = "HTTP Workflows Activities",
+    Name = "HTTP Workflow Activities",
```
Comment (Piedone) [inline src/OrchardCore.Modules/OrchardCore.Workflows/Manifest.cs:22]: "Workflows" as in the module's name?

File: src/OrchardCore.Modules/OrchardCore.Workflows/Manifest.cs [hunk 2]
```diff
@@ -31,4 +31,12 @@
...
     Dependencies = new[] { "OrchardCore.Workflows" },
     Category = "Workflows"
-)]
\ No newline at end of file
+)]
+
+[assembly: Feature(
+    Id = "OrchardCore.Workflows.Session",
+    Name = "Session Workflow Activities",
+    Description = "Provides 'YesSql' Session related activities.",
```
Comment (Piedone) [inline src/OrchardCore.Modules/OrchardCore.Workflows/Manifest.cs:39]:
```suggestion
    Description = "Provides 'YesSql' Session-related activities.",
```

**Review by Piedone — CHANGES_REQUESTED** [review]

File: src/OrchardCore.Modules/OrchardCore.Workflows/Manifest.cs (hunk 1 above)
Comment (jtkech) [reply src/OrchardCore.Modules/OrchardCore.Workflows/Manifest.cs:22]: For HTTP it was "Workflows" but for the Timer it is "Workflow"

    Name = "Timer Workflow Activities",

So i opted to use "Workflow" everywhere "Workflow activities" is used

But as you want

File: src/OrchardCore.Modules/OrchardCore.Workflows/Manifest.cs (hunk 2 above)
Comment (jtkech) [reply src/OrchardCore.Modules/OrchardCore.Workflows/Manifest.cs:39]: Okay

File: src/OrchardCore.Modules/OrchardCore.Workflows/Manifest.cs (hunk 1 above)
Comment (jtkech) [reply src/OrchardCore.Modules/OrchardCore.Workflows/Manifest.cs:22]: Okay, i will put a `s` on both ;)

**Comment by jtkech** [comment]
As a reminder

What we could do is to check if the ambient model state is valid in the Commit Transaction WF activity, it doesn't cover the cases where we update an entity without using the mvc model binding, but at least it would prevent some use cases.

Will see tomorrow

**Comment by jtkech** [comment]
@Piedone 

Done, we now check if the ambient model is valid before committing

**Review by Piedone — APPROVED** [review]
Looks great!

**Pull request merged**